ZuluContent/Skills/SpiritSpeak.cs

```diff
@@ -1,112 +1,245 @@
             {
-                get { return false; }
+                // Fizzle
+                m.FixedEffect(0x3735, 6, 30);
+                m.PlaySound(0x5C);
+                m.SendFailureMessage("You regain your senses as something in the region blocks the spirits energy...");
+                
+                m.Paralyzed = false;
+                return;
             }
 
-            public override double CastDelayFastScalar
+            m.FixedParticles(14089 + Utility.Random(4), 1, 15, 9501, 2100, 4, EffectLayer.Waist);
+            Effects.PlaySound(m.Location, m.Map, 0x107);
+            m.Direction = m.GetDirectionTo(stone);
+            
+            for (var i = 0; i < 3; i++)
             {
-                get { return 0; }
+                m.Animate(0x10D, 7, 3, true, false, 0);
+                await Timer.Pause(1000);
             }
+            
+            var (creatureType, itemType) = UndeadKnowledgeConfig[level];
+            
+            var creature = creatureType.CreateInstance<BaseCreature>();
+            creature.Summoned = true;
+            creature.Blessed = true; // Invulnerable
+            creature.BardImmune = true;
+            creature.MoveToWorld(stone, m.Map);
+            creature.AIObject.Deactivate();
+            creature.PlaySound(creature.GetIdleSound());
+            creature.Direction = creature.GetDirectionTo(m);
+
+            await Timer.Pause(2000);
 
-            public override TimeSpan CastDelayBase
+            creature.PublicOverheadMessage(MessageType.Yell, 33, true, "You dare disturb me in search of knowledge...");
+
+            await Timer.Pause(5000);
+
+            if (!m.ShilCheckSkill(SkillName.SpiritSpeak, level * 8, 0))
             {
-                get { return TimeSpan.FromSeconds(1.0); }
-            }
+                creature.PublicOverheadMessage(MessageType.Yell, 33, true, "I give you only death!");
 
-            public override int GetMana()
+                await Timer.Pause(2000);
+
+                creature.AIObject.Activate();
+                creature.Blessed = false;
+                m.Paralyzed = false;
+                // Give AI time to activate
+                await Timer.Pause(100);
+
+                creature.PlaySound(creature.GetAngerSound());
+                creature.Attack(m);
+            }
+            else
             {
-                return 0;
+                creature.PublicOverheadMessage(MessageType.Yell, 33, true, "I give you only this! Now leave me!");
+
+                var reward = itemType.CreateInstance<Item>();
+                var regs = Reagent.NecroReagents.RandomElement().CreateInstance<Item>();
+                
+                reward.Amount = 1;
+                regs.Amount = level;
+
+                reward.MoveToWorld(stone, m.Map);
+                regs.MoveToWorld(stone, m.Map);
+                
+                await Timer.Pause(5000);
+                creature.Dispel(creature);
```

Comment:
Can this potentially fail? I would maybe just kill the creature since of they are summoned they also won't drop a corpse/loot

Reply:
Essentially does the same thing, the code in `Dispel` is just the dispel effect then `creature.Delete()`.
It's usually used when the `BaseCreature` has `AutoDispel = true` i.e. the dispel summoned creatures on hit.

Reply:
ah ok cool got it